### ISSUE TITLE
Implement Google Assistant for nearest location needing pictures, browsing, uploads, etc

### ISSUE THREAD

**Issue by maskaravivek** [commons-app/apps-android-commons#1514]
Was just wondering if it would be cool to launch the upload flow in the Commons app directly from Google assistant. I can't think much about what all actions could be performed using the assistant launcher but it would surely be a novelty feature to have. 

The basic action that I can think of is: 

"Hey, Google! Upload photo on Commons"

**Comment by misaochan** [comment]
Sounds interesting to me! Would be nice if it would show the nearest location needing pictures, too. :)

**Comment by maskaravivek** [comment]
Yes, great idea. Asking google about nearest location needing pictures would be a much better action. 

Something like:
- find nearest place that needs a picture on Commons
- navigate me to the nearest place that needs a picture on Commons

**Comment by ujjwalagrawal17** [comment]
Also, it would be great to use google Assistant for browsing.
Something like
- Google, search some butterflies from commons.

@maskaravivek can you share a link using which we can integrate Google Assistant?

**Comment by maskaravivek** [comment]
Nice suggestion @ujjwalagrawal17. 

Here's the link for Google Assistant's console. 

https://console.dialogflow.com/api-client/

**Comment by domdomegg** [comment]
This looks really good - I might have a look into it if I get the time
https://codelabs.developers.google.com/codelabs/voice-interaction/index.html
